custom/icds_reports/utils.py

```diff
@@ -176,3 +184,1196 @@ def render_html(self):
+    prev_month_data = get_data_for(config, prev_month)
+
+    return {
+        'kpi': [
+            [
+                {
+                    'label': _('AWC Days Open'),
+                    'help_text': _((
+                        "The total number of days the AWC is open in the given month. The AWC is expected to "
+                        "be open 6 days a week (Not on Sundays and public holidays)")
+                    ),
+                    'percent': percent_increase(
+                        'awc_open',
+                        this_month_data,
+                        prev_month_data,
+                    ),
+                    'value': get_value(this_month_data, 'awc_open'),
+                    'all': '',
+                    'format': 'number'
+                },
+                {
+                    'label': _((
+                        "Percentage of eligible children (ICDS beneficiaries between 0-6 years) "
+                        "who have been weighed in the current month")
+                    ),
+                    'help_text': _('Percentage of AWCs with a functional toilet'),
+                    'percent': percent_diff(
+                        'weighed',
+                        this_month_data,
+                        prev_month_data,
+                        'all'
+                    ),
+                    'value': get_value(this_month_data, 'weighed'),
+                    'all': get_value(this_month_data, 'all'),
+                    'format': 'percent_and_div'
+                }
+            ]
+        ],
+        'charts': [
+            [
+                {
+                    'key': 'AWC Days Open Per Week',
+                    'values': awc_count_chart,
+                    "classed": "dashed",
+                }
+            ],
+            [
+                {
+                    'key': 'PSE- Average Weekly Attendance',
+                    'values': attended_children_chart,
+                    "classed": "dashed",
+                }
+            ]
+        ],
+    }
+
+
+def get_awc_reports_pse(config, month, three_month):
+
+    map_image_data = DailyAttendanceView.objects.filter(
+        pse_date__range=(datetime(*three_month), datetime(*month)), **config
+    ).values('awc_name', 'form_location_lat', 'form_location_long', 'image_name')
+
+    map_data = []
+    image_data = []
+    tmp_image = []
+    img_count = 0
+    count = 1
+    for map_row in map_image_data:
+        lat = map_row['form_location_lat']
+        long = map_row['form_location_long']
+        awc_name = map_row['awc_name']
+        if lat and long:
+            map_data.append({
+                'name': awc_name,
+                'radius': 3,
+                'country': 'IND',
+                'fillKey': 'green',
+                'latitude': lat,
+                'longitude': long
+            })
+        tmp_image.append({'id': count, 'image': 'http://unsplash.it/' + str(300 + count) + '/300'})
```

Comment:
@emord this will be loading data from hq, don't have data locally so for visualisation I added random photos.